feat: 예약, 루틴 관련 기능 (#3)

Lee-sh98 opened the pull request:
## 개요
- #3 **에 대한 domain, service, repository 생성**

## 작업사항
- Equipment, EquipmentGym, Reservation, Routine, EquipmentRoutine에 해당하는 도메인, 서비스, 레포지토리 생성
  - 레포지토리의 경우 JpaRepository를 사용하여 생성, 추후 필요한 메소드 추가예정

Lee-sh98 committed:
feat: add Equipment domain (#3)

Lee-sh98 committed:
feat: add EquipmentGym domain (#3)

Lee-sh98 committed:
feat: add Reservation domain (#3)

Lee-sh98 committed:
feat: add Routine domain (#3)

Lee-sh98 committed:
feat: add Equipment Repository (#3)

Lee-sh98 committed:
feat: add EquipmentGym Repository (#3)

Lee-sh98 committed:
feat: add Reservation Repository (#3)

Lee-sh98 committed:
feat: add Routine Repository (#3)

Lee-sh98 committed:
feat: add EquipmentRoutine Repository (#3)

Lee-sh98 committed:
feat: add Equipment service (#3)

Lee-sh98 committed:
feat: add EquipmentGym service (#3)

Lee-sh98 committed:
feat: add Reservation service (#3)

Lee-sh98 committed:
feat: add Routine service (#3)

Lee-sh98 committed:
feat: add EquipmentRoutine service (#3)

Sangwook02 reviewed (commented):
@Data 모두 @Getter로 수정 바랍니다

Lee-sh98 committed:
feat: apply reviews (#3)

모든 Data 어노테이션을 Getter로 수정
Convert Equipment.time(Long) into duration(Duration)
@JoinColumn의 nullable=false 적용
Enumerated 적용
Routine.count 변수의 final 키워드 삭제

Lee-sh98 committed:
feat: correct JoinColumn in Equipment (#3)

name 추가

Lee-sh98 committed:
Rename: Purpose.part 변수명 변경 (#3)

part -> description

Lee-sh98 committed:
feat: correct ID type in EquipmentRepository (#3)

Sangwook02 reviewed (approved):
변경사항 확인했습니다!

hoyun06 reviewed (approved):
LGTM
